plugins/view-resources/src/components/list/ListCategories.svelte

```diff
@@ -85,7 +85,11 @@
           const f = await getResource(categoryFunc.action)
           const res = hierarchy.clone(await f(_class, query, groupByKey, update, queryId, $statusStore))
           if (res !== undefined) {
-            categories = Array.from(new Set([...categories, ...res]))
+            const set = new Set(categories.map((p) => JSON.stringify(p)))
```

Comment:
Why we need this? They already should be uniq.

Reply:
This is necessary when the function returns a filtered value, but we have an object with a different category field value. It worked fine until we only had the number | string as category, now we have objects (status) and Set can't make it unique.

Reply:
Probably better use something like fastEqual, but I'm not sure

Reply:
categories = await getCategories(client, _class, docs, groupByKey, $statusStore) should always return uniq values already.